test/e2e/performanceprofile/functests/utils/tuned/tuned.go

```diff
@@ -84,38 +84,34 @@ func GetPod(ctx context.Context, node *corev1.Node) (*corev1.Pod, error) {
 }
 
 func WaitForStalldTo(run bool, interval, timeout time.Duration, node *corev1.Node) error {
-	return wait.Poll(interval, timeout, func() (bool, error) {
-		cmd := []string{"/bin/bash", "-c", "pidof stalld || true"}
-		stalldPid, err := nodes.ExecCommandOnNode(cmd, node)
-		if err != nil {
-			return false, fmt.Errorf("failed to execute command %q on node: %q; %w", cmd, node.Name, err)
-		}
+	return wait.PollImmediate(interval, timeout, func() (bool, error) {
+		cmd := []string{"/bin/bash", "-c", "pidof stalld"}
+		stalldPid, _ := nodes.ExecCommandOnNode(cmd, node)
 
-		_, err = strconv.Atoi(stalldPid)
+		_, err := strconv.Atoi(stalldPid)
 		if !run { // we don't want stalld to run
 			if err == nil {
-				return false, fmt.Errorf("node=%q stalld_pid=%q stalld is running when it shouldn't", node.Name, stalldPid)
```

Comment:
Yes this was definitely a mistake since we need to wait some time for TuneD to catch up, but I would replace the error with a warning 
` klog.Warningf("node=%q stalld_pid=%q stalld is still running when it shouldn't", node.Name, stalldPid)`

Reply:
I rather have this manifested as an actual error (see higher level)

Reply:
Then you can returned it at the end of this function instead of write it down multiple times